### PR TITLE
Poly speech/memory fix.

### LINKED ISSUE #79844: Something wrong with Parrot AI
This is causing CI to fail multiple times before randomly working again, similar problem as `monkey_business` unit test

Screenshot (360)

@Jacquerel since AI is your domain maybe you can assess this?

### LINKED ISSUE #79859: Flaky test nuke_cinematic: Parrot As In Repeat Subtree somehow is getting a null potential string while not getting `NO_NEW_PHRASE_AVAILABLE`!
Flaky tests were detected in [this test run](https://github.com/tgstation/tgstation/actions/runs/6925574463/attempts/1). This means that there was a failure that was cleared when the tests were simply restarted.
Failures:
```
nuke_cinematic: [04:45:06] Runtime in code/modules/mob/living/basic/pets/parrot/parrot_ai/parroting_action.dm,18: Parrot As In Repeat Subtree somehow is getting a null potential string while not getting `NO_NEW_PHRASE_AVAILABLE`!
proc name:  stack trace (/proc/_stack_trace)
src: null
call stack:
stack trace("Parrot As In Repeat Subtree so...", "code/modules/mob/living/basic/...", 18)
/datum/ai_planning_subtree/par... (/datum/ai_planning_subtree/parrot_as_in_repeat): SelectBehaviors(/datum/ai_controller/basic_con... (/datum/ai_controller/basic_controller/parrot), 0.5)
/datum/ai_controller/basic_con... (/datum/ai_controller/basic_controller/parrot): SelectBehaviors(0.5)
AI Controller Ticker (/datum/controller/subsystem/ai_controllers): fire(0)
AI Controller Ticker (/datum/controller/subsystem/ai_controllers): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop(2)
Master (/datum/controller/master): StartProcessing(0) at code/modules/mob/living/basic/pets/parrot/parrot_ai/parroting_action.dm:18
```

### LINKED ISSUE #79865: Flaky test fish_rescue_hook: Parrot As In Repeat Subtree somehow is getting a null potential string while not getting `NO_NEW_PHRASE_AVAILABLE`!
Flaky tests were detected in [this test run](https://github.com/tgstation/tgstation/actions/runs/6930470582/attempts/1). This means that there was a failure that was cleared when the tests were simply restarted.
Failures:
```
fish_rescue_hook: [13:09:17] Runtime in code/modules/mob/living/basic/pets/parrot/parrot_ai/parroting_action.dm,18: Parrot As In Repeat Subtree somehow is getting a null potential string while not getting `NO_NEW_PHRASE_AVAILABLE`!
proc name:  stack trace (/proc/_stack_trace)
src: null
call stack:
stack trace("Parrot As In Repeat Subtree so...", "code/modules/mob/living/basic/...", 18)
/datum/ai_planning_subtree/par... (/datum/ai_planning_subtree/parrot_as_in_repeat): SelectBehaviors(/datum/ai_controller/basic_con... (/datum/ai_controller/basic_controller/parrot/ghost), 0.5)
/datum/ai_controller/basic_con... (/datum/ai_controller/basic_controller/parrot/ghost): SelectBehaviors(0.5)
AI Controller Ticker (/datum/controller/subsystem/ai_controllers): fire(0)
AI Controller Ticker (/datum/controller/subsystem/ai_controllers): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop(2)
Master (/datum/controller/master): StartProcessing(0) at code/modules/mob/living/basic/pets/parrot/parrot_ai/parroting_action.dm:18
```

### PR DESCRIPTION
**Pull request by shwb1**
## About The Pull Request
Simple fix for Poly's AI, Read Memory was returning null into a list used for Poly's speech causing a stack error. I changed the returned value to the empty returnable list to negate this problem.

Fixes #79844
Fixes #79859
Fixes #79865

First time making a pull request on Github, please bear with me if i make any mistakes.

## Why It's Good For The Game
Less errors on the servers side. Probably saving the headache of debugging from some other poor soul.

### CODE REVIEW

File: code/modules/mob/living/basic/pets/parrot/poly.dm [hunk 1]
```diff
@@ -129,7 +129,7 @@
 	else
 		var/json_file = file("data/npc_saves/Poly.json")
 		if(!fexists(json_file))
-			return
+			return returnable_list
```
Comment (san7890) [inline code/modules/mob/living/basic/pets/parrot/poly.dm:132]:
```suggestion
			return list()
```

**Review by san7890 — CHANGES_REQUESTED** [review]
Thanks! There is one thing we should do to make this more explicit since we aren't modifying `returnable_list` beforehand just to make it very obvious that we are returning an empty list.

That's really strange how the langserver didnt' catch it, I was under the impression that `RETURN_TYPE(/list)` would lint it somehow - otherwise I would have just handled it like a lazylist or something whenever we invoke `read_memory()`. oh well.

**Comment by SyncIt21** [comment]
Thanks, make sure to link the issue properly in your PR body . Type

> Fixes #79844

And not

>Fix of #79844

**Comment by SyncIt21** [comment]
> Fixes: https://github.com/tgstation/tgstation/issues/79844, https://github.com/tgstation/tgstation/issues/79859

This won't work you should add them as separate lines like so

> Fixes #79844
> Fixes #79859

> about syncing that won't add blank commits/ force a check every time?

It's fine, whenever you make a commit tests will run, it's not under your control, merging with master is by default a blank commit and not an error so it's all good on your end.

**Comment by shwb1** [comment]
Updated. And thanks for the info!

**Pull request merged**